### PR TITLE
APD-176 - create PyPi package

### PR DESCRIPTION
**Pull request by poxip**
https://ydtechnology.atlassian.net/browse/APD-176 - step 1

This also contains **isort** corrects, so that code can pass **--lint** now.

### CODE REVIEW

File: requirements/requirements-base.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+djangorestframework
```
Comment (remik) [inline requirements/requirements-base.txt]: Please define minimum versions.

File: requirements/requirements-base.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+django-rest-swagger
+django-websocket-redis
+redis
+premailer
+
+# push notifications
+django-push-notifications==1.4.1
```
Comment (remik) [inline requirements/requirements-base.txt]: Do not freeze version on pypi packages. It is making a lot problems. This is not recommended pip.

File: runtests.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+sys.path.append(os.path.dirname(__file__))
+
+
+def exit_on_failure(ret, message=None):
+    if ret:
+        sys.exit(ret)
+
+
+def flake8_main(args):
+    print('Running flake8 code linting')
```
Comment (remik) [inline runtests.py]: Please use logger. This will allow to define verbose.

File: runtests.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        print('isort failed: Some modules have incorrectly ordered imports. Fix by running `isort --recursive .`')
+    else:
+        print('isort passed')
+
+    return ret
+
+
+def split_class_and_function(string):
+    class_string, function_string = string.split('.', 1)
+    return "%s and %s" % (class_string, function_string)
```
Comment (remik) [inline runtests.py]: Please use new formatting way https://pyformat.info/

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+language: python
+
+python:
+    - "2.7"
+    - "3.5"
+
+sudo: false
+
+env:
```
Comment (remik) [inline .travis.yml]: Is this only because of `allow_failures` ?

File: runtests.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    # `True` if it looks like a test function is included in the string.
+    return string.startswith('test_') or '.test_' in string
+
+
+def is_class(string):
+    # `True` if first character is uppercase - assume it's a class name.
+    return string[0] == string[0].upper()
+
+
+if __name__ == "__main__":
```
Comment (remik) [inline runtests.py]: Please add some doc/help.

File: setup.cfg [hunk 7]
```diff
@@ -0,0 +1,2 @@
+[metadata]
+description-file = README.md
```
Comment (remik) [inline setup.cfg]: Please add isort if custom.

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+install:
+    # Virtualenv < 14 is required to keep the Python 3.2 builds running.
+    - pip install tox "virtualenv<14"
+
+script:
+    - tox -e $TOX_ENV
+
+after_success:
+    - pip install codecov
+    - codecov -e TOX_ENV
```
Comment (remik) [inline .travis.yml]: Is it send to some service?

File: tox.ini [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+commands = ./runtests.py --fast {posargs} --coverage -rw
+setenv =
+    PYTHONDONTWRITEBYTECODE=1
+    PYTHONWARNINGS=once
+deps =
+    django18: Django==1.8.16
+    django19: Django==1.9.11
+    django110: Django==1.10.3
+    djangomaster: https://github.com/django/django/archive/master.tar.gz
+    -rrequirements/requirements-base.txt
```
Comment (remik) [inline tox.ini]: not missing space?

File: tox.ini [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    {py27,py35}-django{master}
+
+
+[testenv]
+commands = ./runtests.py --fast {posargs} --coverage -rw
+setenv =
+    PYTHONDONTWRITEBYTECODE=1
+    PYTHONWARNINGS=once
+deps =
+    django18: Django==1.8.16
```
Comment (remik) [inline tox.ini]: Do not freeze. Do checking range - otherwise we will need to update it often.

File: tox.ini [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    django19: Django==1.9.11
+    django110: Django==1.10.3
+    djangomaster: https://github.com/django/django/archive/master.tar.gz
+    -rrequirements/requirements-base.txt
+    -rrequirements/requirements-testing.txt
+basepython =
```
Comment (remik) [inline tox.ini]: py34 is still supported so we should add it.

File: tox.ini (hunk 9 above)
Comment (pkrzyzaniak) [inline tox.ini]: nope

File: tox.ini (hunk 10 above)
Comment (pkrzyzaniak) [inline tox.ini]: done

File: tox.ini (hunk 11 above)
Comment (pkrzyzaniak) [inline tox.ini]: done

File: requirements/requirements-base.txt (hunk 1 above)
Comment (pkrzyzaniak) [inline requirements/requirements-base.txt]: done

File: .travis.yml (hunk 8 above)
Comment (pkrzyzaniak) [inline .travis.yml]: codecov.io

File: .travis.yml (hunk 5 above)
Comment (pkrzyzaniak) [inline .travis.yml]: that is just copied from DRF

File: requirements/requirements-base.txt (hunk 2 above)
Comment (pkrzyzaniak) [inline requirements/requirements-base.txt]: done

File: setup.cfg (hunk 7 above)
Comment (pkrzyzaniak) [inline setup.cfg]: it is standard isort (however it was not sorted correctly)

File: requirements/requirements-base.txt (hunk 1 above)
Comment (pkrzyzaniak) [inline requirements/requirements-base.txt]: done

File: runtests.py (hunk 6 above)
Comment (pkrzyzaniak) [inline runtests.py]: ok

File: runtests.py (hunk 4 above)
Comment (pkrzyzaniak) [inline runtests.py]: ok

File: runtests.py (hunk 3 above)
Comment (pkrzyzaniak) [inline runtests.py]: this is only for tox, so I'll skip this one

**Pull request merged**